### ISSUE TITLE
Using lora for training is not effective

### ISSUE THREAD

**Issue by tsrigo** [dhcode-cpp/X-R1#66]
I also tried modifying the format reward; although the value increased, the image remained pretty much the same, oscillating continuously.

![Image](https://github.com/user-attachments/assets/0fc75422-7a03-48c4-876f-7931d418f1f3)

![Image](https://github.com/user-attachments/assets/a48fe05b-ee93-4dda-8f74-e7ebac5c6a99)

My cofig:
```
# Model arguments
model_name_or_path: Qwen/Qwen2.5-7B-Instruct
model_revision: main
torch_dtype: bfloat16
attn_implementation: flash_attention_2

# Data training arguments
dataset_name: xiaodongguaAIGC/X-R1-1500
dataset_configs:
- train

# GRPO trainer config
bf16: true
use_vllm: false
vllm_device: "auto"
vllm_gpu_memory_utilization: 0.45
gradient_accumulation_steps: 16
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false
hub_strategy: every_save
learning_rate: 1.0e-06
log_level: info
logging_steps: 10
logging_strategy: steps
lr_scheduler_type: cosine
max_prompt_length: 256
num_generations: 8
max_completion_length: 1024
max_steps: -1
max_grad_norm: 1.0
num_train_epochs: 5
output_dir: output/X-R1-7B-n8-peft-stable
overwrite_output_dir: true
per_device_train_batch_size: 2
push_to_hub: False
report_to:
  - wandb
save_strategy: "epoch"
seed: 42
warmup_ratio: 0.1

# GRPO trainer config for evaluate
do_eval: true
eval_strategy: "epoch"
per_device_eval_batch_size: 4

# lora config
# task_type: model_args.lora_task_type
# lora_in_4bit: True
lora_r: 16
lora_target_modules: ["q_proj", "v_proj", "k_proj", "embed_tokens"]
lora_alpha: 8
lora_dropout: 0.05
bias: "none"
use_peft: true

```

**Comment by Minami-su** [comment]
me too.

**Comment by Minami-su** [comment]
@tsrigo 请问你的问题解决了吗？使用lora训练grpo好像没有效果，或者效果不明显的问题

**Comment by tsrigo** [comment]
> [@tsrigo](https://github.com/tsrigo) 请问你的问题解决了吗？使用lora训练grpo好像没有效果，或者效果不明显的问题

可以试一下把lora alpha改成32，一般alpha是rank的2-4倍的，不知道他为什么给的0.5倍

以下是用 openr1 跑的，会稍微好一些

<img width="1921" height="960" alt="Image" src="https://github.com/user-attachments/assets/82ef43c3-c325-44d1-bcb7-3f1b9bc1e04c" />

```
# Model arguments
model_name_or_path: Qwen/Qwen2.5-7B-Instruct
model_revision: main
torch_dtype: bfloat16
attn_implementation: flash_attention_2

# Data training arguments
dataset_name: xiaodongguaAIGC/X-R1-1500
dataset_config: default
system_prompt: "A conversation between User and Assistant. The user asks a question, and the Assistant solves it. The assistant first thinks about the reasoning process in the mind and then provides the user with the answer. The reasoning process and answer are enclosed within <think> </think> and <answer> </answer> tags, respectively, i.e., <think> reasoning process here </think><answer> answer here </answer>"

# GRPO trainer config
bf16: true
use_vllm: true
vllm_device: auto
vllm_gpu_memory_utilization: 0.5
do_eval: true
eval_strategy: epoch
eval_steps: 100
gradient_accumulation_steps: 8
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false
hub_model_id: Qwen-2.5-7B-Simple-RL
hub_strategy: every_save
learning_rate: 3.0e-06
log_completions: true
log_level: info
logging_first_step: true
logging_steps: 5
logging_strategy: steps
lr_scheduler_type: cosine
max_prompt_length: 512
max_completion_length: 1024
max_steps: -1
num_generations: 4
num_train_epochs: 5
output_dir: data/Qwen-2.5-7B-n4-r32-alpha64
overwrite_output_dir: true
per_device_eval_batch_size: 4
per_device_train_batch_size: 4
push_to_hub: false
report_to:
- wandb
project: weikai
reward_funcs:
- accuracy
- format
reward_weights:
- 1.0
- 1.0
save_strategy: "epoch"
seed: 42
warmup_ratio: 0.1

# lora config
# task_type: model_args.lora_task_type
# lora_in_4bit: True # no support QLoRA
lora_r: 32 # Reduced from 32 to 16 for more stable training
lora_target_modules: ["q_proj", "v_proj", "k_proj", "embed_tokens"]
lora_alpha: 64
lora_dropout: 0.05 # Added dropout to improve generalization
bias: "none"
use_peft: true
```

**Comment by Minami-su** [comment]
@tsrigo I'm releasing an open-source framework By combining GRPO + QLoRA + DeepSpeed ZeRO-3,https://github.com/Minami-su/deepspeed-grpo-qlora-vllm
